tests/machine.test.ts

```diff
@@ -340,7 +341,7 @@ describe(OlmMachine.name, () => {
                         signed_curve25519: 20,
                     },
                 });
-                const marked = await m.markRequestAsSent(request.id, request.type, hypothetical_response);
+                const marked = await m.markRequestAsSent(request.id!, request.type, hypothetical_response);
```

Comment:
Yes, exactly so. Or rather, it is telling Typescript that we know that, in practice, `request.id` will be defined even though `SignatureUploadRequest.id` is optional, so that it doesn't complain when we pass it into `markRequestAsSent` which requires a `string` rather than a `string|undefined`.